src/libraries/Directory.Build.props

```diff
@@ -49,7 +49,7 @@
     <Nullable Condition="'$(Nullable)' == '' and '$(IsTestProject)' != 'true'">enable</Nullable>
     <Nullable Condition="'$(Nullable)' == '' and '$(IsTestProject)' == 'true'">annotations</Nullable>
     <!-- EnableAOTAnalyzer is enabled by default for src projects. -->
-    <EnableAOTAnalyzer Condition="'$(EnableAOTAnalyzer)' == '' and '$(IsSourceProject)' == 'true'">true</EnableAOTAnalyzer>
+    <EnableAOTAnalyzer Condition="'$(EnableAOTAnalyzer)' == '' and '$(IsSourceProject)' == 'true' and $([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net7.0'))">true</EnableAOTAnalyzer>
```

Comment:
A general principle is to not condition on `TargetFramework` in props files as taht only works in multi-targeting projects for which the inner project build has the TargetFramework available as a global property. For single non-multi targeting builds that's not true and TargetFramework isn't available in props.

Consider moving that to the D.B.targets file.